lib/install/phases/coredns.go

```diff
@@ -93,12 +93,24 @@ func (r *corednsExecutor) Execute(ctx context.Context) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	servers := resolvConf.Servers
+
+	// Optionally try and load upstream nameservers from systemd-resolved by reading the compatibility resolv.conf
+	// More Info: https://github.com/gravitational/gravity/issues/606#issuecomment-529171440
+	// TODO(knisbet) is there a better way to pull upstream resolvers directly from systemd?
+	systemdResolvConf, err := systeminfo.ResolvFromFile("/run/systemd/resolve/resolv.conf")
```

Comment:
Even though I also have `systemd-resolved` active on my ubuntu `18.04`, I don't have any upstream nameservers listed in `/run/systemd/resolve/resolv.conf` - instead it just links back to `systemd-resolved` with `127.0.0.53`.
I'm not sure which of the [compatibility modes](https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html#/etc/resolv.conf) is on but the file does not list any link-specific DNS servers.

One way I saw this information being queried from `systemd-resolved` is by fetching `DNS` servers property for the specific link from [DBus](https://www.freedesktop.org/wiki/Software/systemd/resolved/). Here's an [example](https://github.com/FiloSottile/captive-browser/blob/master/cmd/systemd-networkd-dns/main.go) of the query.
Also, on my system the `resolve1` DBus service was not registered and I subsequently failed to query it with DBus although `systemd-resolve --status` is reporting the DNS servers properly.

Reply:
Well that sucks, I was hoping regardless of compatibility mode it was writing that file. thanks for checking. Guess we'll have to go the dbus route, but the problem I have with that is I found out how to pull the per interface DNS over dbug (although that go code is a good reference), I didn't see a clear indication on how to pull the global resolver when set...

```
Global
         DNS Servers: 8.8.8.8
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
```

`systemd-resolved --status` obviously pulls it from somewhere, just need to waste time figuring out where that is.

Reply:
Ok, so after restarting the `NetworkManager` service on my system, the `systemd-resolved` did update the `/run/systemd/resolve/resolv.conf` file with the list of external nameservers and seems to keep them in sync if they're updated in the settings.